src/core/QCheck_ng.ml

```diff
@@ -0,0 +1,1966 @@
+(*
+QCheck: Random testing for OCaml
+copyright (c) 2013-2017, Guillaume Bury, Simon Cruanes, Vincent Hugot, Jan Midtgaard
+all rights reserved.
+*)
+
+(** {1 Quickcheck inspired property-based testing} *)
+
+let poly_compare=compare
+open Printf
+
+module RS = Random.State
+
+let (|>) x f = f x
+
+let rec foldn ~f ~init:acc i =
+  if i = 0 then acc else foldn ~f ~init:(f acc i) (i-1)
+
+let _is_some = function Some _ -> true | None -> false
+
+let _opt_map_2 ~f a b = match a, b with
+  | Some x, Some y -> Some (f x y)
+  | _ -> None
+
+let _opt_map_3 ~f a b c = match a, b, c with
+  | Some x, Some y, Some z -> Some (f x y z)
+  | _ -> None
+
+let _opt_map_4 ~f a b c d = match a, b, c, d with
+  | Some x, Some y, Some z, Some w -> Some (f x y z w)
+  | _ -> None
+
+let _opt_sum a b = match a, b with
+  | Some _, _ -> a
+  | None, _ -> b
+
+let sum_int = List.fold_left (+) 0
+
+exception FailedPrecondition
+(* raised if precondition is false *)
+
+exception No_example_found of string
```

Comment:
It is a little bit weird to see two kind of cases. `WordWord` vs `Word_word` (I suggest to pick the second one).

Reply:
Agreed, I took the liberty (I hope @c-cube doesn't mind/take offense :sweat_smile: ) to align case for all exceptions to `Word_word`